org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/UnresolvedElementsSubProcessor.java

```diff
@@ -550,6 +550,23 @@ private static boolean hasFieldWithName(ITypeBinding typeBinding, String name) {
 
 	private static int evauateTypeKind(ASTNode node, IJavaProject project) {
 		int kind = ASTResolving.getPossibleTypeKinds(node, JavaModelUtil.is50OrHigher(project));
+
+		/*
+		 *  TODO : This code block should be contributed to ASTResolving.getPossibleTypeKinds(..)
```

Comment:
Note : Attempting to upstream this as https://git.eclipse.org/r/c/jdt/eclipse.jdt.ui/+/170568 .

Reply:
When M2 is released, this block may be deleted ( https://bugs.eclipse.org/bugs/show_bug.cgi?id=567770 )